packages/serverless-component/serverless.js

```diff
@@ -237,22 +239,46 @@ class NextjsComponent extends Component {
 
     const defaultEdgeLambdaPublishOutputs = await defaultEdgeLambda.publishVersion();
 
-    const defaultCloudfrontInputs =
-      (inputs.cloudfront && inputs.cloudfront.defaults) || {};
+    let defaultCloudfrontInputs;
+    if (inputs.cloudfront && inputs.cloudfront.defaults) {
+      defaultCloudfrontInputs = inputs.cloudfront.defaults;
+      delete inputs.cloudfront.defaults;
+    } else {
+      defaultCloudfrontInputs = {};
+    }
+
+    // Add any custom cloudfront configuration
+    // this includes overrides for _next, static and api
+    Object.entries(cloudFrontConfigs).map(([path, config]) => {
```

Comment:
Before running these cloudFrontConfig overrides, could you validate that the path pattern provided actually matches a Next.js page please? The `defaultBuildManifest` already contains the information about which pages there are. [Here is an example](https://github.com/danielcondemarin/serverless-next.js/blob/master/packages/lambda-at-edge/tests/fixtures/default-build-manifest.json) of the manifest structure.

Reply:
One more thing, we should also validate users can't add something like an `/api/whatever` path here because there is a separate cache behaviour and origin request function for `api/*` paths.

Reply:
this was nontrivial. I've added a validation helper and some test cases. have a look

Reply:
Looks good thanks!